invenio_workflows/api.py

```diff
@@ -210,15 +210,52 @@ def __repr__(self):
             return self.model.__repr__()
 
     def __eq__(self, other):
-        """Enable equal operators on WorkflowObjects."""
+        """Enable equal operators on WorkflowObjects.
+
+        ..Note: The task history timestamps are not taken into account.
+        """
+        def _are_same_task(task1, task2):
+            task1 = dict(task1)
+            task1.pop('time')
+            task2 = dict(task2)
+            task2.pop('time')
+
+            return (task1 == task2)
+
+        def _are_same_task_history(task_history1, task_history2):
+            for task1, task2 in zip(task_history1, task_history2):
+                if not _are_same_task(task1, task2):
+                    return False
+
+                return True
+
+        def _have_same_extra_data(wflw1, wflw2):
+            for key, value1 in wflw1.extra_data.items():
```

Comment:
Missing the other way around, if there's a key in wflw2.extra_data that's not in wflw1.extra_data